src/Store.lua

```diff
@@ -49,19 +52,21 @@ function Store.new(reducer, initialState, middlewares, errorReporter)
 
 	local self = {}
 
-	self._errorReporter = errorReporter or defaultErrorReporter
+	self._errorReporter = errorReporter or rethrowErrorReporter
 	self._isDispatching = false
 	self._reducer = reducer
 	local initAction = {
 		type = "@@INIT",
 	}
-	self._lastAction = initAction
-	local ok, result = pcall(function()
+	self._actionLog = { initAction }
+	local ok, result = xpcall(function()
 		self._state = reducer(initialState, initAction)
-	end)
+	end, tracebackReporter)
 	if not ok then
-		local message = ("Caught error with init action of reducer (%s): %s"):format(tostring(reducer), tostring(result))
-		errorReporter:reportErrorImmediately(message, debug.traceback())
+		self._errorReporter.reportReducerError(initialState, initAction, {
+			message = "Caught error in reducer with init",
+			thrownValue = result,
+		})
 		self._state = initialState
```

Comment:
I know this is from previous code, but is this state update logic ok? If "not ok then self._state and self._lastState both equal initialState"? If we're resetting, should lastState be the previous lastState?

Reply:
It shouldn't be an issue; technically, both things are true. Your last state was the initial state passed into `Store.new`; your current state is _also_ that, because your reducer threw an error.